frappe/desk/page/user_profile/user_profile.py

```diff
@@ -77,6 +78,9 @@ def update_profile_info(profile_info):
 	profile_info = frappe.parse_json(profile_info)
 	keys = ["location", "interest", "user_image", "bio"]
 
+	if "user_image" in profile_info:
+		profile_info["user_image"] = escape_html(profile_info["user_image"])
```

Comment:
This should be used where it's rendered. Not while storing.

Reply:
E.g. storing *anything* is not unsafe, using it incorrectly by injecting it as HTML inside the DOM is.